mono/metadata/threads.c

```diff
@@ -1709,6 +1709,13 @@ mono_thread_internal_detach (MonoThread *thread)
 		return;
 	MONO_ENTER_GC_UNSAFE;
 	mono_thread_detach_internal (thread->internal_thread);
+
+	// Not really a background change, but break the wait in a
+	// thread calling mono_thread_manage_internal in case it is
+	// waiting for this thread.
+	MONO_ENTER_GC_SAFE;
+	mono_os_event_set (&background_change_event);
```

Comment:
In the case of aborting threads looks like mono_thread_manage_internal doesn't wait on this event.

Reply:
hm. I'm not sure why this commit is here.  I don't remember what this was for. I'll revert this bit.